Add GOVUK radios [part 5]

tombye opened the pull request:
Part 5 of migrating our radio buttons to use the [GOV.UK Frontend radios component](https://design-system.service.gov.uk/components/radios/).

https://www.pivotaltracker.com/story/show/170030262

This pull request adds a `GovukRadiosFieldWithNoneOption` class, sets `SetEmailBranding.branding_style` to use it and updates all the pages/routes that use the `SetEmailBranding` and `SetLetterBranding` forms.

### Notes

There is an existing version of `GovukRadiosFieldWithNoneOption` called `RadioFieldWithNoneOption` which could have been converted instead. `RadioFieldWithNoneOption` is used by some fields I intend to change in future so I'm leaving that work for then.

Part 1 of this work was done in https://github.com/alphagov/notifications-admin/pull/3715.
Pull requests for the other parts are:
- part 2: https://github.com/alphagov/notifications-admin/pull/3727
- part 3: https://github.com/alphagov/notifications-admin/pull/3728
- part 4: https://github.com/alphagov/notifications-admin/pull/3730

Pull requests in this stream of work don't need to be completed in any order.

klssmith commented:
This change seems to have stopped the preview page (for both emails and letters) from appearing.

klssmith commented on app/templates/views/service-settings/set-letter-branding.html:
```diff
@@ -31,7 +30,7 @@
           label='Search by name',
           autofocus=True
         ) }}
-        {{ radios(form.branding_style, hide_legend=True) }}
```
Not having `hide_legend` here or on the other letter branding form means that we now show the text "Branding style" above the radio buttons. That might be fine though - I'm not sure if there was a reason we were showing that for emails but not for letters.

tombye replied:
I looked into this and both the org and service settings pages for email branding show the legend but those for letter branding don't. I can't see a reason why they should differ as they're basically the same interface.

It seems sensible to make them consistent and, speaking personally, I'd choose to hide the legend as that information is available in the page title and h1 any hiding it adds vertical space.

What do you think?

klssmith replied:
I think the page looks better with the legend hidden since it duplicates existing information on the page. I can't think of a reason why we'd hide it for letters but show it for emails, so I'd probably go with just removing it for all branding types.

tombye replied:
Great. I've made a change in the form that serves all these pages to hide the legend so marking this as resolved.

tombye commented:
> This change seems to have stopped the preview page (for both emails and letters) from appearing.

Great spot! 🥇 I totally missed that changing the radios HTML in the branding pages broke a CSS selector in the JS for the preview iframe. Looks like it failed silently so no tests to say this happened.

I've updated that selector and updated the bits of the JS tests that render radios HTML to match the new style.

klssmith commented:
The preview pane is back 🎉. One more thing I spotted is that as well as changing the preview pane so that it looks for `.govuk-radios__item` instead of `.multiple-choice input`, you'll also have to make a similar change for the `target_selector` of the `live_search` macro.

tombye commented:
> The preview pane is back 🎉. One more thing I spotted is that as well as changing the preview pane so that it looks for `.govuk-radios__item` instead of `.multiple-choice input`, you'll also have to make a similar change for the `target_selector` of the `live_search` macro.

Good point! I've updated that selector on all the pages changed. I also updated the tests for live-search so any radios HTML they use is the new version.
